include/openenclave/seal.h

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ *
+ * This file defines constants and structures for sealing APIs.
+ *
+ * Only TEE agnostic definitions should go in this file. SGX specific
+ * definitions should go in sgx/seal.h
+ *
+ */
+#ifndef _OE_SEAL_H
+#define _OE_SEAL_H
+
+#include <openenclave/bits/evidence.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * Seal settings as TLV tuples.
+ *
+ * It's strongly recommended to use OE_SEAL_SET_* helper macros to set up this
+ * structure, rather than assigning to members directly.
+ */
+typedef struct _oe_seal_setting
+{
+    int type;      ///< Setting type. See oe_seal_setting_type_t for details.
+    uint32_t size; ///< Size of the buffer pointed to by \c value.p
+    union {
+        uint64_t q; ///< quad-word value. \c size should be set to \c 0.
+        uint32_t d; ///< double-word value. \c size should be set to \c 0.
+        uint16_t w; ///< word value. \c size should be set to \c 0.
+        uint8_t b;  ///< byte value. \c size should be set to \c 0.
+        const void*
+            p; ///< buffer. \c size should be set to the buffer size in bytes.
+    } value;
+} oe_seal_setting_t;
+
+/*
+ * TEE agnostic seal settings supported by OE SDK
+ */
+enum oe_seal_setting_type_t
+{
+    /**
+     * Desired sealing policy - either \c OE_SEAL_POLICY_UNIQUE or \c
+     * OE_SEAL_POLICY_PRODUCT.
+     */
+    OE_SEAL_SETTING_POLICY,
+
+    /**
+     * Additional context to be included in seal key derivation. Please note
+     * that not every seal plug-in supports this setting, in which case
+     * \coe_seal() will return \c OE_UNSUPPORTED. It's highly recommended NOT
+     * to specify this setting.
+     */
+    OE_SEAL_SETTING_ADDITIONAL_CONTEXT,
+
+    /**
+     * IV (Initial Vector) to use should the underlying cipher requires one.
+     * It's highly recommended NOT to specify this setting.
+     */
+    OE_SEAL_SETTING_IV,
+
+    /**
+     * Upper bound of seal setting types.
+     *
+     * No seal setting types shall be defined at or above this.
+     */
+    OE_SEAL_SETTING_MAX = 0x10000,
+
+    /**
+     * Lower bound of plug-in defined setting types.
+     *
+     * All values below this are reserved by OpenEnclave SDK.
+     */
+    OE_SEAL_SETTING_PLUGIN_DEFINED = OE_SEAL_SETTING_MAX / 2
+};
+
+#define __OE_SEAL_SET_POINTER(t, p, s) \
+    {                                  \
+        (t), s,                        \
+        {                              \
+            (uint64_t)(p)              \
+        }                              \
+    }
+#define __OE_SEAL_SET_VALUE(t, v) __OE_SEAL_SET_POINTER(t, v, 0)
+
+/**
+ * Initialize a \c oe_seal_setting_t structure to specify seal policy.
+ *
+ * @param w Should be either \c OE_SEAL_POLICY_UNIQUE or \c
```

Comment:
```suggestion
 * @param[in] w Should be either \c OE_SEAL_POLICY_UNIQUE or \c
```

Please use doxygen [in] or [out] or [in,out] annotation on every @param (including lines 107, 120, etc.)